tests/functional/test_proxy.py

```diff
@@ -24,3 +33,49 @@ def test_minimal_proxy(ethereum, minimal_proxy_container, chain, owner):
     if isinstance(abi, list):
         assert abi == []
     # else: is messed up from other test (xdist).
+
+
+def test_provider_not_supports_get_storage(
+    get_contract_type, owner, vyper_contract_instance, ethereum, chain, networks
+):
+    """
+    The get storage slot RPC is required to detect this proxy, so it won't work
+    on EthTester provider. However, we can make sure that it doesn't try to
+    call `get_storage()` more than once.
+    """
+
+    class MyProvider(LocalProvider):
```

Comment:
I had to make a new provider because EthTester supports getting storage now, which is a good thing